Svc/TraceFileLogger/TraceFileLogger.cpp

```diff
@@ -0,0 +1,242 @@
+
+    // ----------------------------------------------------------------------
+    // Handler implementations for user-defined typed input ports
+    // ----------------------------------------------------------------------
+
+    void TraceFileLogger::TraceBufferLogger_handler(FwIndexType portNum,
+                                                FwTraceIdType id,
+                                                Fw::Time& timeTag,
+                                                const Fw::TraceCfg::TraceType& type,
+                                                Fw::TraceBuffer& args) {
+            
+            FW_ASSERT(type.isValid());
+            U16 bit_mask = static_cast<U16> (FILTER_BIT << type.e);
+            U32 traceSize = 0;
+            
+            //Only log trace types that are enabled by either config or user
+            if(!(this->m_traceFilter & bit_mask)) {
+                //TODO: Should we generate an event here, letting user know that this specific filter is disabled?
+                return;
+            }
+            //Only log trace Ids that are not in the list 
+            if(this->filterTraceId.search_array(id,nullptr) == true) {
+                return;
+            }
+           
+           //Make a call to reset
+           Fw::SerializeBufferBase& buf_ref = m_file_buffer.getSerializeRepr();
+           buf_ref.resetSer();
+           (void)buf_ref.serialize(id);
+           (void)buf_ref.serialize(timeTag);
+           if(FW_TRACE_RECORD_MINIMAL == false) {
+            (void)buf_ref.serialize(args);
+            traceSize = m_file_buffer.getSize(); //Record max size of each trace record for circular file
+           }
+           else {
+            traceSize = buf_ref.getBuffLength(); //Record only id & timetag
+           }
+           //Note: Because its a circular file we're writing the full buffer capacity to the file
+           //      instead of the actual buffer size (variable based on number of args). This will 
+           //      ensure when the file is overwritten, we preserve older records
+           this->write_log_file(m_file_buffer.getData(),traceSize);
+           // If we choose not to use circular file write then use below instead. 
+           //this->write_log_file(m_file_buffer.getData(),buf_ref.getBuffLength());
+            
+    }
+
+
+    // ----------------------------------------------------------------------
+    // Handler implementations for commands
+    // ----------------------------------------------------------------------
+
+    void TraceFileLogger ::EnableTrace_cmdHandler(FwOpcodeType opCode, U32 cmdSeq, Svc::TraceFileLogger_Enable enable) {
+        
+        FW_ASSERT(enable.isValid());
+        this->m_enable_trace = enable; 
+        this->cmdResponse_out(opCode, cmdSeq, Fw::CmdResponse::OK);
+    }
+
+    void TraceFileLogger ::DumpTraceDp_cmdHandler(FwOpcodeType opCode, U32 cmdSeq) {
+        this->cmdResponse_out(opCode, cmdSeq, Fw::CmdResponse::OK);
```

Comment:
If we are writing a wrapping file, we don't need this?

Reply:
I was thinking about that, it was part of the initial implementation, and I wasn't sure if we should keep it or not.